### PR TITLE
Added ROS2 Github action

### PR DESCRIPTION
**Pull request by dhananjay-patki**
Same as previous PR

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    branches:
+      - rr_devel
+  workflow_dispatch:
+jobs:
+  ROS2_build_test:
+    runs-on: "${{ matrix.os }}"
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        os:
```
Comment (triccyx) [inline .github/workflows/main.yml:16]:
```yam
 strategy:
      matrix:
        build_type: [Release, Debug]
        os: [ubuntu-latest, macOS-latest, windows-latest]
```
I leave windows and Mac only to show the syntax. In the case, here, we can add old ubuntu version.
I would keep at least Release/Debug

File: .github/workflows/ros2_build_test/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+colcon build --continue-on-error --packages-skip alica_ros_proxy alica_tracing alica_ros_turtlesim alica_tests supplementary_tests alica_msgs
+
+# colcon test
+
+# colcon test-result --verbose
+
+
+if [ $? -eq 0 ]
+  then exit 0
```
Comment (triccyx) [inline .github/workflows/ros2_build_test/entrypoint.sh]: Perhaps here we can put an error log.

**Review by triccyx — COMMENTED** [review]
Just suggestions no errors for me

File: .github/workflows/main.yml (hunk 1 above)
Comment (Dale-Koenig) [reply .github/workflows/main.yml:16]: No point doing multiple os because we build in docker anyway

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: Build and Test
+on:
+  push:
+    branches:
+      - rr_devel
+  pull_request:
+    branches:
+      - rr_devel
```
Comment (Dale-Koenig) [inline .github/workflows/main.yml]: Update to new branch name

**Review by triccyx — APPROVED** [review]

File: .github/workflows/main.yml (hunk 3 above)
Comment (dhananjay-patki) [reply .github/workflows/main.yml]: Updated!

**Review by Dale-Koenig — APPROVED** [review]

File: .github/workflows/ros2_build_test/entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+. /opt/ros/humble/setup.sh
+
+colcon build --continue-on-error --packages-skip alica_ros_proxy alica_tracing alica_ros_turtlesim alica_tests supplementary_tests alica_msgs
+
+# colcon test
+
+# colcon test-result --verbose
+
+
+if [ $? -eq 0 ]
```
Comment (Dale-Koenig) [inline .github/workflows/ros2_build_test/entrypoint.sh]: Note that when we change this, you have to retrieve or save $? immediately after the colcon build command, so uncommenting could cause problems.  Another choice is to do `colcon build <stuff> || exit 1`

Comment (dhananjay-patki) [reply .github/workflows/ros2_build_test/entrypoint.sh]: Modified it to that, it does make more sense that way

**Pull request merged**